### PR TITLE
Package collection model updates

### PR DESCRIPTION
**Pull request by yim-lee**
Requires https://github.com/apple/swift-package-manager/pull/3122

### CODE REVIEW

File: PackageCollectionFormats/v1.md [hunk 1]
```diff
@@ -67,26 +70,20 @@ A version object has metadata extracted from `Package.swift` and optionally addi
...
 
-* `verifiedPlatforms`: An array of platforms in which the package version has been tested and verified. Valid platform names include `macOS`, `iOS`, `tvOS`, `watchOS`, `Linux`, `Android`, and `Windows`. This is different from `platforms` in `Package.swift`. **Optional.**
+* `verifiedCompatibility`: An array of compatible platforms and Swift versions that has been tested and verified for. Valid platform names include `macOS`, `iOS`, `tvOS`, `watchOS`, `Linux`, `Android`, and `Windows`. Swift version should be semantic version string and as specific as possible. **Optional.**
 
 ```json
 {
-  "name": "macOS"
+  "platform": {
+    "name": "macOS"
+  },
```
Comment (finestructure) [inline PackageCollectionFormats/v1.md:79]: Hi Yim, while updating our package collection code to the latest model I noticed `Platform.name` and `PlatformVersion.name` as `String`.

Are these fields intended for display purposes only or also to be machine read? If the latter, I was wondering if it'd be better to use

```
enum PlatformName: String, Codable {
  case macOS
  ...
}
```

to ensure case/spelling is uniform.

Comment (finestructure) [reply PackageCollectionFormats/v1.md:79]: Then again that'd require all platforms to be defined beforehand - which I suppose is not really feasible. Never mind :)

Comment (yim-lee) [reply PackageCollectionFormats/v1.md:79]: Sorry @finestructure just saw these comments. Both `PlatformVersion.name` and `Platform.name` get converted into `PackageModel.Platform` [code](https://github.com/apple/swift-package-manager/blob/main/Sources/PackageCollections/Providers/JSONPackageCollectionProvider.swift#L207), which is an enum. It's best-effort--if no match is found the corresponding entry will be dropped.

**Comment by yim-lee** [comment]
@swift-ci test

**Pull request merged**